support-services/src/main/scala/com/gu/support/promotions/PromotionCache.scala

```diff
@@ -5,18 +5,16 @@ import org.joda.time.{DateTime, Minutes}
 
 import scala.concurrent.stm.{Ref, atomic}
 
-case class PromotionCache(fetched: DateTime, promotions: Iterable[Promotion]) {
-  def isFresh: Boolean = DateTime.now.getMillis - fetched.getMillis < PromotionCache.maxAge
+case class PromotionCacheResponse(fetched: DateTime, promotions: Iterable[Promotion]) {
+  def isFresh: Boolean = DateTime.now.getMillis - fetched.getMillis < Minutes.ONE.toStandardDuration.getMillis
 }
 
-object PromotionCache {
-  val maxAge = Minutes.ONE.toStandardDuration.getMillis
-
-  val promotionsRef = Ref[Option[PromotionCache]](None)
+class PromotionCache {
+  val promotionsRef = Ref[Option[PromotionCacheResponse]](None)
```

Comment:
As far as I know this isn't mutable, but rather a `Ref[T]` that contains `val t: T`. And you can change that using a method on `Ref` that returns a new `Ref` containing another value for `t` (but still of type `T`). 

And it get's stored in memory.

Reply:
I don't claim to understand fully, but I checked the documentation here
https://nbronson.github.io/scala-stm/#:~:text=ScalaSTM%20provides%20a%20mutable%20cell%20called%20a%20Ref

and they call it a "mutable cell"

So the knowledge of what's in the container is mutable, rather than the container itself or the thing that is inside it.